nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java

```diff
@@ -84,24 +82,15 @@ public UploadResponse handle(Repository repository, ComponentUpload upload) thro
     PartPayload payload = upload.getAssetUploads().get(0).getPayload();
 
     String fileName = payload.getName() != null ? payload.getName() : StringUtils.EMPTY;
+    AssetKind assetKind = AssetKind.getAssetKindByFileName(fileName);
 
-    try (TempBlob tempBlob = storageFacet.createTempBlob(payload, HASH_ALGORITHMS)) {
-      HelmAttributes attributesFromInputStream;
-      AssetKind assetKind;
-      String extension;
-
-      if (fileName.endsWith(PROVENANCE_EXTENSION)) {
-        attributesFromInputStream = helmPackageParser.getAttributesProvenanceFromInputStream(tempBlob.get());
-        assetKind = AssetKind.HELM_PROVENANCE;
-        extension = PROVENANCE_EXTENSION;
-      } else if (fileName.endsWith(TGZ_EXTENSION)) {
-        attributesFromInputStream = helmPackageParser.getAttributesFromInputStream(tempBlob.get());
-        assetKind = AssetKind.HELM_PACKAGE;
-        extension = TGZ_EXTENSION;
-      } else {
-        throw new IllegalArgumentException("Unsupported extension. Extension must be .tgz or .tgz.prov");
-      }
+    if (assetKind != AssetKind.HELM_PROVENANCE && assetKind != AssetKind.HELM_PACKAGE) {
+      throw new IllegalArgumentException("Unsupported extension. Extension must be .tgz or .tgz.prov");
```

Comment:
Asset kind is set in recipe by verifying extension. You don't check extension here, only asset kind so message should be something like: "Unsupported asset kind".

Reply:
Done

Reply:
Wrong place. It's upload UI handler, not recipe handler. Reverting...